ExtractData/PKHeX.ExtractData/ExtTMHM3.cs

```diff
@@ -0,0 +1,376 @@
+            bool FR_LG_EqualsComp = ExtractUtils.CheckArrayEqual(FR_Compatibility, LG_Compatibility);
+            bool R_E_EqualsComp   = ExtractUtils.CheckArrayEqual(R_Compatibility, E_Compatibility);
+            bool FR_E_EqualsComp  = ExtractUtils.CheckArrayEqual(FR_Compatibility, E_Compatibility);
+
+            /*From here only Emerald data is used*/
+
+            TraceTMHMList(log_folder + "G3_HMTM.log", E_TMHM, TMMoves_Gen3.Length, HMMoves_Gen3.Length);
+            TraceTMHMCompatibilityList(log_folder, "G3_HMTM_Index_Comp.log"    , E_TMHM, E_FlagsCompatibility, TMMoves_Gen3.Length, HMMoves_Gen3.Length,true);
+
+            /*Data is stored in gen 3 index number, not in dex number. 
+            * Order the data in dex order to have the same order in pkhex like the personal table*/
+            E_Compatibility = GData.TransformDexOrder(E_Compatibility);
+            E_FlagsCompatibility = RawFlagsToBoolean(E_Compatibility, TMMoves_Gen3.Length + HMMoves_Gen3.Length);
+
+            TraceTMHMCompatibilityList(log_folder, "G3_HMTM_DexNumber_Comp.log", E_TMHM, E_FlagsCompatibility, TMMoves_Gen3.Length, HMMoves_Gen3.Length, false);
+
+            TMHMTutorMoves[] E_MovesCompatibility = TMHMTutorMoves.getArray(E_Compatibility);
+
+            byte[] TutorsFileData = ExtractUtils.PackTMHMTutorMoves("g3", E_MovesCompatibility, GBA_HMTMComp_Size);
+            Directory.CreateDirectory(output_folder);
+            File.WriteAllBytes(output_folder + "hmtm_g3.pkl", TutorsFileData);
+
+            //Test the fileS are correct
+            TMHMTutorMoves[] Tutors_Check_Data = TMHMTutorMoves.getArray(ExtractUtils.unpackMini(File.ReadAllBytes(output_folder + "hmtm_g3.pkl"), "g3"));
+            string diff = string.Empty;
+            bool check = ExtractUtils.CheckEquals(E_MovesCompatibility, Tutors_Check_Data, out diff);
+            if (!check)
+                throw new Exception("Check HM TM Moves Gen 3." + diff);
+
+        }
+
+
+        public int[][] InheritanceTMHMMOves(bool[][] FlagsCompatibility, ushort[] MoveList)
+        {
+            int[][] Output_Moves = new int[FlagsCompatibility.Length][];
+
+            for (int i = 0; i < FlagsCompatibility.Length; i++)
+            {
+                Output_Moves[i] = MoveList.
+                            Select((move, index) => new { index = index, move = move }).
+                            Where(m => FlagsCompatibility[i][m.index]).
+                            Select(m => (int)m.move).
+                            ToArray();
+            }
+
+            return Output_Moves;
+        }
+
+        internal void TraceTMHMList(string filename, ushort[] MoveList, int TMCount, int HMCount)
+        {
+            string log = string.Empty;
+            string[] all_moves = Util.getMovesList("en");
+
+            for (int i = 0; i < TMCount; i++)
+            {
+                log += $"TM{(i + 1).ToString("00")} Move {MoveList[i]} { all_moves[MoveList[i]]}" + Environment.NewLine;
```

Comment:
I didnt thought about performance hit because this program only need to be executed once to extract the data. I did it and the data is already in the PKHex folder, i could just commit the pkl files but i thought no one could trust the data if i dont show how i extrated it from the roms